src/consumeUntil.js

```diff
@@ -129,14 +129,16 @@ export default async function consumeUntil (target_content_height, container, op
 				let remaining_height = target_content_height - nodes.height;
 				let empty_lines = remaining_height / lh;
 
-				// TODO use CSS widows and orphans rather than these hardcoded values
-				if (empty_lines >= 2) {
+				let orphaned_lines = style.orphans;
+				let widowed_lines = style.widows;
+
+				if (empty_lines >= orphaned_lines) {
 					let child_height = util.getHeight(child, {force: true});
 					let child_lines = child_height / lh;
 
-					if (child_lines >= 3.99) {
+					if (child_lines >= widowed_lines - .01) {
 						child.normalize();
-						let consumed = await consumeUntil(Math.min(remaining_height, child_height - 2 * lh), child, options);
+						let consumed = await consumeUntil(Math.min(remaining_height, child_height - orphaned_lines * lh), child, options);
```

Comment:
I think it's widows you want here, not orphans. Orphans are at the bottom of the page, i.e. the first fragment, widows are the top of a page, i.e. the second fragment.